src/game/client/components/menus.cpp

```diff
@@ -2681,12 +2695,25 @@ void CMenus::OnStateChange(int NewState, int OldState)
 		m_DownloadLastCheckTime = time_get();
 		m_DownloadLastCheckSize = 0;
 		m_DownloadSpeed = 0.0f;
-		//client_serverinfo_request();
+
+		// restore password from favorites
+		const char *pPassword = ServerBrowser()->GetFavoritePassword(g_Config.m_UiServerAddress);
```

Comment:
The global password shouldn't be overwritten when you fetch a password from a favorite entry.